This fixes a regression introduced in 1.8.0 where libssh2 fails to build
with the gcrypt backend.

Upstream bug URL:

https://github.com/libssh2/libssh2/issues/150

Patch copied from upstream source repository:

https://github.com/libssh2/libssh2/commit/ced924b78a40126606797ef57a74066eb3b4b83f

From ced924b78a40126606797ef57a74066eb3b4b83f Mon Sep 17 00:00:00 2001
From: Sergei Trofimovich <siarheit@google.com>
Date: Mon, 31 Oct 2016 09:04:33 +0000
Subject: [PATCH] acinclude.m4: fix ./configure --with-libgcrypt

diff --git a/acinclude.m4 b/acinclude.m4
index 734ef07..c78260c 100644
--- a/acinclude.m4
+++ b/acinclude.m4
@@ -412,9 +412,9 @@ AC_DEFUN([LIBSSH2_CHECKFOR_GCRYPT], [
 
   old_LDFLAGS=$LDFLAGS
   old_CFLAGS=$CFLAGS
-  if test -n "$use_libgcrypt" && test "$use_libgcrypt" != "no"; then
-    LDFLAGS="$LDFLAGS -L$use_libgcrypt/lib"
-    CFLAGS="$CFLAGS -I$use_libgcrypt/include"
+  if test -n "$with_libgcrypt_prefix" && test "$use_libgcrypt" != "no"; then
+    LDFLAGS="$LDFLAGS -L$with_libgcrypt_prefix/lib"
+    CFLAGS="$CFLAGS -I$with_libgcrypt_prefix/include"
   fi
   AC_LIB_HAVE_LINKFLAGS([gcrypt], [], [
     #include <gcrypt.h>
